patch-package
--- a/node_modules/truffle-core/lib/command.js
+++ b/node_modules/truffle-core/lib/command.js
@@ -108,9 +108,9 @@ Command.prototype.run = function(inputStrings, options, callback) {

 Command.prototype.displayGeneralHelp = function() {
   this.args
-    .usage("Truffle v" + (version.bundle || version.core) + " - a development framework for Ethereum"
+    .usage("Truffle+Analyze v" + (version.bundle || version.core) + " - a development framework for Ethereum"
     + OS.EOL + OS.EOL
-    + "Usage: truffle <command> [options]")
+    + "Usage: truffle+analyze <command> [options]")
     .epilog("See more at http://truffleframework.com/docs")
     .showHelp();
 }
new file mode 100644
--- /dev/null
+++ b/node_modules/truffle-core/lib/commands/analyze.js
@@ -0,0 +1,30 @@
+const TA = require("../../../truffle-analyze");
+var command = {
+  command: 'analyze',
+  description: 'Run MythOS analyses on a contract',
+  help: {
+    usage: `truffle+analyze analyze [options]`,
+
+    options: [
+      // {
+      //   option: '--mode {full | quick}',
+      //   description: "Set analysis mode to be either quick (fast) or full (in depth)",
+      // },
+      {
+        option: '--debug',
+        description: 'Provide additional debug output',
+      },{
+        option: `--style {stylish | unix | visualstudio | table | tap | ...}`,
+        description: 'Report format in given es-lint style style.\n' +
+          '                    See https://eslint.org/docs/user-guide/formatters/ for a full list.'
+      },{
+        option: '--timeout *seconds* ',
+        description: 'Limit MythOS analysis time to *s* seconds.\n' +
+          '                    The default is 30 seconds.',
+      },
+    ]
+  },
+  run: TA.run
+}
+
+module.exports = command;
--- a/node_modules/truffle-core/lib/commands/compile.js
+++ b/node_modules/truffle-core/lib/commands/compile.js
@@ -21,7 +21,7 @@ var command = {
     }
   },
   help: {
-    usage: "truffle compile [--list <filter>] [--all] [--network <name>]",
+    usage: "truffle+analyze compile [--list <filter>] [--all] [--network <name>]",
     options: [
       {
         option: "--all",
--- a/node_modules/truffle-core/lib/commands/console.js
+++ b/node_modules/truffle-core/lib/commands/console.js
@@ -3,7 +3,7 @@ var command = {
   description: 'Run a console with contract abstractions and commands available',
   builder: {},
   help: {
-    usage: "truffle console [--network <name>] [--verbose-rpc]",
+    usage: "truffle+analyze console [--network <name>] [--verbose-rpc]",
     options: [
       {
         option: "--network <name>",
--- a/node_modules/truffle-core/lib/commands/create.js
+++ b/node_modules/truffle-core/lib/commands/create.js
@@ -12,7 +12,7 @@ var command = {
     }
   },
   help: {
-    usage: "truffle create <artifact_type> <ArtifactName>",
+    usage: "truffle+analyze create <artifact_type> <ArtifactName>",
     options: [
       {
         option: "<artifact_type>",
--- a/node_modules/truffle-core/lib/commands/debug.js
+++ b/node_modules/truffle-core/lib/commands/debug.js
@@ -7,7 +7,7 @@ var command = {
     }
   },
   help: {
-    usage: "truffle debug <transaction_hash>",
+    usage: "truffle+analyze debug <transaction_hash>",
     options: [
       {
         option: "<transaction_hash>",
--- a/node_modules/truffle-core/lib/commands/deploy.js
+++ b/node_modules/truffle-core/lib/commands/deploy.js
@@ -5,7 +5,7 @@ var command = {
   description: '(alias for migrate)',
   builder: migrate.builder,
   help: {
-    usage: "truffle deploy [--reset] [-f <number>] [--network <name>] [--compile-all] [--verbose-rpc]",
+    usage: "truffle+analyze deploy [--reset] [-f <number>] [--network <name>] [--compile-all] [--verbose-rpc]",
     options: migrate.help.options,
   },
   run: migrate.run
--- a/node_modules/truffle-core/lib/commands/develop.js
+++ b/node_modules/truffle-core/lib/commands/develop.js
@@ -11,7 +11,7 @@ var command = {
     }
   },
   help: {
-    usage: "truffle develop",
+    usage: "truffle+analyze develop",
     options: [],
   },
   runConsole: function(config, testrpcOptions, done) {
--- a/node_modules/truffle-core/lib/commands/exec.js
+++ b/node_modules/truffle-core/lib/commands/exec.js
@@ -15,7 +15,7 @@ var command = {
     }
   },
   help: {
-    usage: "truffle exec <script.js> [--network <name>]",
+    usage: "truffle+analyze exec <script.js> [--network <name>]",
     options: [
       {
         option: "<script.js>",
--- a/node_modules/truffle-core/lib/commands/help.js
+++ b/node_modules/truffle-core/lib/commands/help.js
@@ -2,7 +2,7 @@ var command = {
   command: "help",
   description: "Display a list of all commands or information about a specific command",
   help: {
-    usage: "truffle help [<command>]",
+    usage: "truffle+analyze help [<command>]",
     options: [
       {
         option: "<command>",
--- a/node_modules/truffle-core/lib/commands/index.js
+++ b/node_modules/truffle-core/lib/commands/index.js
@@ -4,6 +4,7 @@ module.exports = {
   migrate: require("./migrate"),
   deploy: require("./deploy"),
   test: require("./test"),
+  analyze: require("./analyze"),
   debug: require("./debug"),
   opcode: require("./opcode"),
   console: require("./console"),
--- a/node_modules/truffle-core/lib/commands/init.js
+++ b/node_modules/truffle-core/lib/commands/init.js
@@ -3,7 +3,7 @@ var command = {
   description: 'Initialize new and empty Ethereum project',
   builder: {},
   help: {
-    usage: "truffle init [--force]",
+    usage: "truffle+analyze init [--force]",
     options: [
       {
         option: "--force",
--- a/node_modules/truffle-core/lib/commands/install.js
+++ b/node_modules/truffle-core/lib/commands/install.js
@@ -3,7 +3,7 @@ var command = {
   description: 'Install a package from the Ethereum Package Registry',
   builder: {},
   help: {
-    usage: "truffle install <package_name>[@<version>]",
+    usage: "truffle+analyze install <package_name>[@<version>]",
     options: [
       {
         option: "package_name",
--- a/node_modules/truffle-core/lib/commands/migrate.js
+++ b/node_modules/truffle-core/lib/commands/migrate.js
@@ -27,7 +27,7 @@ var command = {
     },
   },
   help: {
-    usage: "truffle migrate [--reset] [-f <number>] [--network <name>] [--compile-all] [--verbose-rpc] [--interactive]",
+    usage: "truffle+analyze migrate [--reset] [-f <number>] [--network <name>] [--compile-all] [--verbose-rpc] [--interactive]",
     options: [
       {
         option: "--reset",
--- a/node_modules/truffle-core/lib/commands/networks.js
+++ b/node_modules/truffle-core/lib/commands/networks.js
@@ -9,7 +9,7 @@ var command = {
     }
   },
   help: {
-    usage: "truffle networks [--clean]",
+    usage: "truffle+analyze networks [--clean]",
     options: [
       {
         option: "--clean",
--- a/node_modules/truffle-core/lib/commands/opcode.js
+++ b/node_modules/truffle-core/lib/commands/opcode.js
@@ -8,7 +8,7 @@ var command = {
     }
   },
   help: {
-    usage: "truffle opcode <contract_name>",
+    usage: "truffle+analyze opcode <contract_name>",
     options: [
       {
         option: "<contract_name>",
--- a/node_modules/truffle-core/lib/commands/publish.js
+++ b/node_modules/truffle-core/lib/commands/publish.js
@@ -3,7 +3,7 @@ var command = {
   description: 'Publish a package to the Ethereum Package Registry',
   builder: {},
   help: {
-    usage: "truffle publish",
+    usage: "truffle+analyze publish",
     options: [],
   },
   run: function (options, done) {
--- a/node_modules/truffle-core/lib/commands/test.js
+++ b/node_modules/truffle-core/lib/commands/test.js
@@ -9,7 +9,7 @@ var command = {
     },
   },
   help: {
-    usage: "truffle test [<test_file>] [--compile-all] [--network <name>] [--verbose-rpc]",
+    usage: "truffle+analyze test [<test_file>] [--compile-all] [--network <name>] [--verbose-rpc]",
     options: [
       {
         option: "<test_file>",
--- a/node_modules/truffle-core/lib/commands/unbox.js
+++ b/node_modules/truffle-core/lib/commands/unbox.js
@@ -52,7 +52,7 @@ var command = {
   description: 'Download a Truffle Box, a pre-built Truffle project',
   builder: {},
   help: {
-    usage: "truffle unbox [<box_name>] [--force]",
+    usage: "truffle+analyze unbox [<box_name>] [--force]",
     options: [
       {
         option: "<box_name>",
--- a/node_modules/truffle-core/lib/commands/version.js
+++ b/node_modules/truffle-core/lib/commands/version.js
@@ -3,7 +3,7 @@ var command = {
   description: 'Show version number and exit',
   builder: {},
   help: {
-    usage: "truffle version",
+    usage: "truffle+analyze version",
     options: [],
   },
   run: function (options, done) {
@@ -17,7 +17,7 @@ var command = {
       bundle_version = "(unbundled)";
     }

-    options.logger.log("Truffle " + bundle_version + " (core: " + version.core + ")");
+    options.logger.log("Truffle+Analyze " + bundle_version + " (core: " + version.core + ")");
     options.logger.log("Solidity v" + version.solc + " (solc-js)");

     done();
